services/user.js

```diff
@@ -0,0 +1,35 @@
+const jwt = require("jsonwebtoken");
+
+//function to create JWT token
+const createToken = (/*userId*/) => {
+  const token = jwt.sign(
+    {
+      /* user id from mongo DB */
+    },
+    process.env.JWT_SECRET,
+    { expiresIn: process.env.JWT_EXPIRES }
+  );
+  return token;
+};
+
+const loginUser = async (userData) => {
+  // make DB query
+  // check if we have user in DB
+  // compare bcrypted passwords
+
+  const token = createToken(/* userId*/);
+
+  return { token /* user */ };
+};
+
+const registerUser = async (userData) => {
+  //check if emails is not registered already in DB
+
+  const token = createToken(/* userId*/);
+  return { token /* user */ };
```

Comment:
Why you return token on register? Register should create new user. Then user should use loginUser to get token

Reply:
in loginUser i have different functionality, or the functionality that i dont use on registering. like when i log in i check in DB if i have that user than i have to compare passwords create token and return it to make user logged in, while when user registers, i create new user document in mongo collection i encrypt password and then also create token and return it to make user logged in. am i not correct in that logic?

Reply:
i use registerUser function on api/user/register route and loginUser on api/user/login route

Reply:
you mean that when user fills the register form and presses the 'register' button then user should go to login page fill the login from to login?

Reply:
`you mean that when user fills the registration form and presses the 'register' button then user should go to login page fill the login form to login`
Yes. Register endpoint should only create user and login endpoint should only generate token (single responsibility principle)